src/Yarhl/IO/DataStream.cs

```diff
@@ -497,19 +492,102 @@ public void WriteTo(DataStream stream)
                 throw new ObjectDisposedException(nameof(stream));
 
             long currPos = Position;
-            Seek(0, SeekMode.Start);
+            Seek(start, SeekMode.Start);
 
             const int BufferSize = 70 * 1024;
             byte[] buffer = new byte[Length > BufferSize ? BufferSize : Length];
 
-            while (!EndOfStream) {
+            while (!EndOfStream)
+            {
                 int read = BlockRead(this, buffer);
                 stream.Write(buffer, 0, read);
             }
 
             Seek(currPos, SeekMode.Start);
         }
 
+        /// <summary>
+        /// Writes a defined length stream into another DataStream starting in a defined position.
+        /// </summary>
+        /// <param name="start">Defined starting position.</param>
+        /// <param name="length">Defined length to be written.</param>
+        /// <param name="stream">Output DataStream.</param>
+        public void WriteSegmentTo(long start, int length, DataStream stream)
+        {
+            if (Disposed)
+                throw new ObjectDisposedException(nameof(DataStream));
+
+            if (stream == null)
+                throw new ArgumentNullException(nameof(stream));
+            if (stream.Disposed)
+                throw new ObjectDisposedException(nameof(stream));
+
+            long currPos = Position;
+            Seek(start, SeekMode.Start);
+
+            byte[] buffer = new byte[length];
```

Comment:
This may cause memory issues if length is too big. For instance imagine calling this method for a big CVM file to remove the header. We should do the same as WriteSegmentTo without length does. In fact, what about if we adapt that logic here and that method calls this one with length = `Length`?